SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "GenerateTestMapsCommandlet.h"
+#include "TestMaps/GeneratedTestMap.h"
+
+DEFINE_LOG_CATEGORY(LogGenerateTestMapsCommandlet);
+
+UGenerateTestMapsCommandlet::UGenerateTestMapsCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+}
+
+int32 UGenerateTestMapsCommandlet::Main(const FString& CmdLineParams)
+{
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Generate test maps commandlet started."));
+
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Deleting the %s folder."), *UGeneratedTestMap::GetGeneratedMapFolder());
+	if (FPaths::DirectoryExists(UGeneratedTestMap::GetGeneratedMapFolder()))
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		PlatformFile.DeleteDirectoryRecursively(*UGeneratedTestMap::GetGeneratedMapFolder());
+	}
+
+	// Have to gather the classes first and then iterate over the copy, because creating a map triggers a GC which can modify the object
+	// array this is iterating through
+	TArray<UClass*> TestMapClasses;
+	for (TObjectIterator<UClass> Iter; Iter; ++Iter)
+	{
+		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) && *Iter != UGeneratedTestMap::StaticClass())
```

Comment:
what is this check for?

Reply:
The condition in the if?
I am looking for all classes that Unreal knows about, which are children of the UGeneratedTestMap class, and they are not the UGeneratedTestMap class itself.